### PR TITLE
Feature/ansonlee99/1011099544 sidebar UI

### PR DESCRIPTION
**Pull request by AnsonLee99**
### Administrative Info
Monday Board ID: 1011099544
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Monday Item ID]-[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
Added sidebar to the website. Items can be expanded individually or by clicking on "expand-all". Items can also be sorted either alphabetically or by expiration date. 

One thing to note: when sorting by expiration date, the plus sign on the item header seems a bit out-of-place since the item cannot be expanded. However, I kept the plus sign there since that's what is shown on the Figma design. I also removed the "expand-all" button since it served no purpose when sorting by expiration date.

Disclaimer: This PR also includes some changes to add_button.js since I moved that file to its own directory. 

### Testing
Clicked around on the headers to verify they expanded/collapsed correctly. Changed the "sort-by" dropdown to check values are being correctly sorted.

### Confirmation of Change 
Sort-by: Alphabetical
image
----------------------
Sort-by: Expiry
image

### CODE REVIEW

File: client/src/home/sidebar/Sidebar.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+  const sortAlpha = (a, b) => {
+    return a.name > b.name;
+  }
+
+  const sortExpiry = (a, b) => {
+    a = new Date(a.outDate);
+    b = new Date(b.outDate);
+    let diff = a.getTime() - b.getTime();
+    if (diff != 0) {
```
Comment (petabite) [inline client/src/home/sidebar/Sidebar.js]: should this be a `!==` ?
```suggestion
    if (diff !== 0) {
```

**Review by petabite — APPROVED** [review]
Nicee! Works perfectly. Yeah, good point, we should probably bring up the plus icon being there when sorting by expiration to the design team since that is a little confusing.

**Review by hahsu — CHANGES_REQUESTED** [review]
Nice work! Sidebar.js and SidebarItem.js are both pretty lengthy files. Can you add inline comments to better describe what each chunk of code is doing?

**Comment by AnsonLee99** [comment]
> Nice work! Sidebar.js and SidebarItem.js are both pretty lengthy files. Can you add inline comments to better describe what each chunk of code is doing?

Done!

**Review by hahsu — APPROVED** [review]
lgtm!

**Pull request merged**